src/mtcp/mtcp_restart.c

```diff
@@ -657,6 +664,14 @@ restorememoryareas(RestoreInfo *rinfo_ptr)
 
   DPRINTF("close cpfd %d\n", restore_info.fd);
   mtcp_sys_close(restore_info.fd);
+  double readTime = 0.0;
+#ifdef TIMING
+  struct timeval endValue;
+  mtcp_sys_gettimeofday(&endValue, NULL);
+  struct timeval diff;
+  timersub(&endValue, &restore_info.startValue, &diff);
+  readTime = diff.tv_sec + (diff.tv_usec / 1000000.0);
+#endif
```

Comment:
This is not obligatory, but I think I see an opportunity for re-factoring here.  I'm guessing that the above pattern will happen a lot in our code.  Should we create a new function and call it inside the `ifdef`?